src/what_is_an_ABI.md

```diff
@@ -32,3 +32,13 @@ To quote from the initial proposal:
 > TODO: more recent examples.
 
 ## What would a stable modular ABI look like for Rust?
+
+- **Fully specified** - There should be no ambiguous cases, either in how to map Rust to the ABI, or how the ABI maps back to Rust objects.
+- **Versioned** - There should be a mechanism to determine which version of the ABI a chunk of code is using by querying the code itself.  The version should follow [semver](https://semver.org/) conventions, and should be both machine and human parseable, so that linkers and loaders can decide if two pieces of code are compatible (from an ABI standpoint).  Under some circumstances, it may be possible for a linker or loader to generate code to bind chunks of code together that have different ABI versions.
```

Comment:
Should the Semver of the ABI be separate from or a part of the Semver of the package? i.e. should a non-breaking change that breaks the ABI API be considered a breaking change? How will this be enforced? Again, this might not be the place to address this, but it's important to keep in mind.

Reply:
I intended for them to be separated.  The package's API version should be able to change entirely independently of the ABI specification.  The ABI version is so that ABI tools (linkers, loaders, debuggers, decompilers, etc., etc., etc.) can make sense of the binary blob that they are inspecting, or if they know that they can't (can't handle the given ABI version), then they can return a sensible error to the user.